lib/src/merged_tree.rs

```diff
@@ -204,6 +204,9 @@ impl MergedTree {
             }
             Ok(_) => Ok(None),
             Err(merge) => {
+                if !merge.is_tree() {
+                    return Ok(None);
+                }
                 let trees = merge
                     .try_map_async(|value| async move {
                         match value {
```

Comment:
nit: insert `Some(_) => unreachable!()` to this match to clarify that the value should be either `Some(Tree(_))` or `None`?